src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/processorders/ProcessOrdersAboutToSubmitHandler.java

```diff
@@ -49,65 +53,91 @@ public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType
     @Override
     public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(FinremCallbackRequest callbackRequest,
                                                                               String userAuthorisation) {
+        FinremCaseData caseData = callbackRequest.getCaseDetails().getData();
+        // handleNewDocument is required to be done before calling super method which stamps the uploading document.
+        handleNewDocument(caseData);
         GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> resp = super.handle(callbackRequest, userAuthorisation);
```

Comment:
I think it would make more sense not to inherit from another handler and just copy the small bit of functionality into this class instead.